cpp/ql/test/experimental/query-tests/Security/CWE/CWE-691/semmle/tests/test.c

```diff
@@ -0,0 +1,28 @@
+int tmpFunction(){
+  return 5;
+}
+void workFunction_0(char *s) {
+  int intSize;
+  char buf[80];
+  if(intSize>0 && intSize<80 && memset(buf,0,intSize)) return; // GOOD
+  if(intSize>0 & intSize<80 & memset(buf,0,intSize)) return; // BAD
+  if(intSize>0 && tmpFunction()) return;
+  if(intSize<0 & tmpFunction()) return; // BAD
+}
+void workFunction_1(char *s) {
+  int intA,intB;
+
+  if(intA + intB) return; // BAD
```

Comment:
I'm not so sure about this test annotation. Two things:
1. Is this really bad? I think this is an okay way of testing whether the sum is zero. If you agree with this you should change this comment to say `// GOOD`.
2. The query isn't currently flagging this. If you think the query really should raise an alert here, the comment should say `// BAD [NOT DETECTED]`.